src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs

```diff
@@ -389,21 +445,157 @@ private IEnumerable<IServiceConnection> CreateFixedServiceConnection(int count)
             {
                 yield return CreateServiceConnectionCore(InitialConnectionType);
             }
-        }
-
+        }
+
+        private async Task WriteServiceStatusPingAsync()
+        {
+            await WriteAsync(RuntimeServicePingMessage.GetStatusPingMessage(true));
+        }
+
+        private async Task WriteServerIdsPingAsync()
+        {
+            await WriteAsync(RuntimeServicePingMessage.GetServersPingMessage());
+            if (Stopwatch.GetTimestamp() - _serverIdContext.Item2 > DefaultServersPingTimeoutTicks)
+            {
+                // reset value if expired.
+                _serverIdContext = DefaultServerIdContext;
+            }
+        }
+
+        private CustomizedPingTimer CreateServersPingTimer()
+        {
+            return new CustomizedPingTimer(Logger, Constants.CustomizedPingTimer.Servers, WriteServerIdsPingAsync, DefaultServersPingInterval, DefaultServersPingInterval);
+        }
+
+        private sealed class CustomizedPingTimer : IDisposable
+        {
+            private long _lastSendTimestamp = 0;
+            private readonly long _defaultPingTicks;
+
+            private readonly string _pingName;
+            private readonly Func<Task> _writePing;
+            private readonly TimeSpan _dueTime;
+            private readonly TimeSpan _intervalTime;
+
+            // Considering parallel add endpoints to save time
+            // Add a counter control multiple time call Start() and Stop() correctly
+            private volatile int _counter = 0;
+
+            private TimerAwaitable _timer;
+            private ILogger _logger { get; }
+
+            public CustomizedPingTimer(ILogger logger, string pingName, Func<Task> writePing, TimeSpan dueTime, TimeSpan intervalTime)
+            {
+                _logger = logger;
+                _pingName = pingName;
+                _writePing = writePing;
+                _dueTime = dueTime;
+                _intervalTime = intervalTime;
+                _defaultPingTicks = intervalTime.Seconds * Stopwatch.Frequency;
+
+                _timer = Init();
+            }
+
+            public void Start()
+            {
+                if (_counter == 0)
+                {
+                    _timer.Start();
+                    _ = PingAsync(_timer);
+                }
+                Interlocked.Increment(ref _counter);
+            }
+
+            public void Stop()
+            {
+                if (_counter == 0)
```

Comment:
also here, 

```cs
if (Interlocked.Decrement(ref _counter) == 0)
{...}

```

Reply:
Considering avoid wrong stop to break further operation, use long type to be able to use Interlocked.Read( ) for check first.